diff -r dcc15836c1f3 src/lens.c
--- a/src/lens.c	Mon Oct 30 16:10:01 2017 +0100
+++ b/src/lens.c	Thu Nov 09 21:00:06 2017 +0100
@@ -1287,6 +1287,8 @@
 {
     uint32_t raw = *(uint32_t *) buf;
 
+    lens_info.raw_iso_ae=raw;
+    
     #if defined(FRAME_ISO)
     if (lv && is_movie_mode()) raw = (uint8_t)FRAME_ISO;
     #endif
@@ -1389,6 +1391,7 @@
 {
     /* this gets updated in Av mode (where PROP_SHUTTER is not updated); same for P, Auto and so on */
     /* it becomes 0 when camera is no longer metering */
+    lens_info.raw_shutter_ae=buf[0];
     
     if (shooting_mode == SHOOTMODE_M || shooting_mode == SHOOTMODE_TV)
     {
@@ -2339,6 +2342,15 @@
 }
 #endif
 
+
+int read_frame_iso()
+{
+    #ifdef FRAME_ISO
+        return (int)(FRAME_ISO  & 0xFF);
+    #else
+        return 0;
+    #endif
+}
 /** Camera control functions */
 int lens_set_rawaperture( int aperture)
 {
diff -r dcc15836c1f3 src/lens.h
--- a/src/lens.h	Mon Oct 30 16:10:01 2017 +0100
+++ b/src/lens.h	Thu Nov 09 21:00:06 2017 +0100
@@ -67,6 +67,8 @@
         // raw exposure values, in 1/8 EV steps
         uint8_t                 raw_aperture;
         uint8_t                 raw_shutter;
+        uint8_t                 raw_shutter_ae;   
+        uint8_t                 raw_iso_ae;            
         uint8_t                 raw_iso;
         uint8_t                 raw_iso_auto;
         uint8_t                 raw_picstyle;           /* fixme: move it out */
@@ -210,6 +212,7 @@
 extern int bv_set_rawshutter(unsigned shutter);
 extern int bv_set_rawaperture(unsigned aperture);
 extern int bv_set_rawiso(unsigned iso);
+extern int read_frame_iso();
 
 /* private, to be refactored */
 extern void bv_update();
